src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/GeneralApplicationHelper.java

```diff
@@ -0,0 +1,217 @@
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.GENERAL_APPLICATION_TRACKING;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class GeneralApplicationHelper {
+
+    private final ObjectMapper objectMapper;
+
+    public List<GeneralApplicationCollectionData> getGeneralApplicationList(Map<String, Object> caseData) {
+        return Optional.ofNullable(caseData.get(GENERAL_APPLICATION_COLLECTION))
+            .map(this::covertToGeneralApplicationData).orElse(new ArrayList<>());
+
+    }
+
+    public List<GeneralApplicationCollectionData> getReadyForRejectOrReadyForReferList(Map<String, Object> caseData) {
+        return getGeneralApplicationList(caseData).stream()
+            .filter(obj -> Objects.equals(obj.getGeneralApplicationItems().getGeneralApplicationStatus(), CREATED.getId()))
+            .toList();
+    }
+
+    public List<GeneralApplicationCollectionData> getReferredList(Map<String, Object> caseData) {
+        return getGeneralApplicationList(caseData).stream()
+            .filter(obj -> Objects.equals(obj.getGeneralApplicationItems().getGeneralApplicationStatus(), REFERRED.getId()))
+            .toList();
+    }
+
+    public List<GeneralApplicationCollectionData> getOutcomeList(Map<String, Object> caseData) {
+        return getGeneralApplicationList(caseData).stream()
+            .filter(this::isEquals)
+            .toList();
+    }
+
+    private boolean isEquals(GeneralApplicationCollectionData obj) {
+        String generalApplicationStatus = obj.getGeneralApplicationItems().getGeneralApplicationStatus();
+        return (Objects.equals(generalApplicationStatus, APPROVED.getId())
+            || Objects.equals(generalApplicationStatus, NOT_APPROVED.getId())
+            || Objects.equals(generalApplicationStatus, OTHER.getId()));
+    }
+
+
+    public List<GeneralApplicationCollectionData> covertToGeneralApplicationData(Object object) {
+        return objectMapper.registerModule(new JavaTimeModule()).convertValue(object, new TypeReference<>() {
+        });
+    }
+
+    public List<GeneralApplicationData> convertToGeneralApplicationDataList(Object object) {
+        return objectMapper.convertValue(object, new TypeReference<>() {
+        });
+    }
+
+    public CaseDocument convertToCaseDocument(Object object) {
+        if (object != null) {
+            return objectMapper.registerModule(new JavaTimeModule()).convertValue(object, CaseDocument.class);
+        }
+        return null;
+    }
+
+    public String objectToString(Object object) {
+        if (object != null) {
+            return objectMapper.registerModule(new JavaTimeModule()).convertValue(object, String.class);
+        }
+        return null;
+    }
+
+    public LocalDate objectToDateTime(Object object) {
+        if (object != null) {
+            return objectMapper.registerModule(new JavaTimeModule()).convertValue(object, LocalDate.class);
```

Comment:
Object mapper looks like it isn't needed -  `LocalDate.parse(Objects.toString(myObj))` instead perhaps?

Reply:
in this too many cascading - if you want I can change